django/utils/functional.py

```diff
@@ -18,13 +22,26 @@ class cached_property:
     Decorator that converts a method with a single self argument into a
     property cached on the instance.
 
-    Optional ``name`` argument allows you to make cached properties of other
-    methods. (e.g.  url = cached_property(get_absolute_url, name='url') )
+    It is possible to make a cached property out of an existing method:
+    (e.g.  ``url = cached_property(get_absolute_url)`` ).
+    On Python versions before 3.6, you'll need to provide the optional ``name``
+    argument. (e.g.  ``url = cached_property(get_absolute_url, name='url')`` ).
     """
     def __init__(self, func, name=None):
+        func_name = func.__name__
+        if PY35 and not name and re.match(r'^__.*(?<!__)$', func_name):
```

Comment:
Can you just do `name.startswith('__') and not name.endswith('__')`? Simpler is better